docs/modules/ROOT/pages/reference/exoscale-osbapi.adoc

```diff
@@ -0,0 +1,368 @@
+
+=== Suspension Flow
+
+This flow is triggered when an Exoscale organization:
+
+* changes their current plan
+* is suspended
+* changes the user list on Exoscale side and user sync is turned on
+
+The suspension uses a special "suspension" plan.
+
+[source,json]
+----
+PATCH http://exo-osbapi.vshn.net/v2/service_instances/:instance_id
+
+{
+    "service_id": "service-test-guid",
+    "plan_id": "plan1-test-guid", <1>
+    "parameters": {
+        "users": [
+            {
+                "email":"email",
+                "full_name": "full name",
+                "role":"owner|tech"
+            }
+        ]
+    }
+}
+----
+<1> Special suspension plan, to be defined
+
+Sources:
+
+* https://community.exoscale.com/documentation/vendor/marketplace-managed-services-provision/#service-instance-update[Exoscale docs - Service Instance Update^]
+* https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#updating-a-service-instance[OSB API Spec^]
+
+When the suspension plan is triggered, we send an E-Mail to customers@vshn.ch with all the information we have, so that we can check back with Exoscale what to do.
+No service is automatically suspended. If it has to happen, we'll do it manually.
+
+Also, the affected `OSBServiceInstance` is updated.
+
+=== Offboarding Flow
+
+This flow is triggered when an Exoscale organization:
+
+* decides to unsubscribe the product
+* suspension is not resolved before 7 days in trial mode, or 30 days outside of trial mode, which triggers a purge of their resources
+* decides to close their Exoscale account, or their account is terminated
+
+[source,json]
+----
+DELETE http://exo-osbapi.vshn.net/v2/service_instances/:instance_id?service_id=service-test-guid&plan_id=plan1-test-guid
+----
+
+Sources:
+
+* https://community.exoscale.com/documentation/vendor/marketplace-managed-services-provision/#deprovisioning[Exoscale docs - Deprovisioning^]
+* https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#deprovisioning[OSB API Spec^]
+
+We send an E-Mail for each service instance which gets deleted that way, telling the customer that the service has to be removed from the VSHN Portal.
+See also <<Deprovisioning>>, which talks about the details.
+
+The matching `OSBServiceInstance` gets deleted.
+When all `OSBServiceInstance` objects are deleted, an email is sent to customer@vshn.ch for the final closure of the organization.
+
+Also, there is a monitoring check which triggers when no `OSBServiceInstance` is available, but service instances are still there.
+This means the customer didn't delete the service instances.
+
+* TODO: Should we automatically delete service instances after a grace period when there is no `OSBServiceInstance` available?
```

Comment:
IMHO we should set the `OSBServiceInstance` as an owner reference on all relevant objects.